lms/views/decorators/h_api.py

```diff
@@ -40,12 +40,16 @@ def wrapper(context, request):
         }
 
         try:
-            hapi_svc.patch(
-                f"users/{context.h_username}", {"display_name": context.h_display_name}
-            )
-        except HAPINotFoundError:
-            # Call the h API to create the user in h if it doesn't exist already.
-            hapi_svc.post("users", user_data)
+            try:
+                hapi_svc.patch(
+                    f"users/{context.h_username}",
+                    {"display_name": context.h_display_name},
+                )
+            except HAPINotFoundError:
+                # Call the h API to create the user in h if it doesn't exist already.
+                hapi_svc.post("users", user_data)
+        except HAPIError as err:
+            raise HTTPInternalServerError(explanation=err.explanation) from err
```

Comment:
The nested `try`'s here are unfortunate, but the solution is to move the "Try PATCH and if it 404s try POST" (a.k.a upsert) logic into the service rather than having it here in the view, and that's outside of the scope of this PR